azure-pipelines.yml

```diff
@@ -23,3 +23,104 @@ jobs:
+      # made the process crash as it couldn't find deactivate
+      #echo '##vso[task.prependpath]$HOME/miniconda3/bin'
+      echo '##vso[task.setvariable variable=CONDA_ENV]pandas-dev'
+      echo '##vso[task.setvariable variable=ENV_FILE]environment.yml'
+      echo '##vso[task.setvariable variable=AZURE]true'
+    displayName: 'Setting environment variables'
+
+  # Do not require a conda environment
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      ci/code_checks.sh patterns
+    displayName: 'Looking for unwanted patterns'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      sudo apt-get install -y libc6-dev-i386
+      ci/incremental/install_miniconda.sh
+      ci/incremental/setup_conda_environment.sh
+    displayName: 'Set up environment'
+
+  # Do not require pandas
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh lint
+    displayName: 'Linting'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh dependencies
+    displayName: 'Dependencies consistency'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/incremental/build.sh
+    displayName: 'Build'
+    condition: true
+
+  # Require pandas
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh code
+    displayName: 'Checks on imported code'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh doctests
+    displayName: 'Running doctests'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      ci/code_checks.sh docstrings
+    displayName: 'Docstring validation'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      pytest --capture=no --strict scripts
+    displayName: 'Testing docstring validaton script'
+    condition: true
+
+  - script: |
+      export PATH=$HOME/miniconda3/bin:$PATH
+      source activate pandas-dev
+      git remote add upstream https://github.com/pandas-dev/pandas.git
+      git fetch upstream
+      if git diff upstream/master --name-only | grep -q "^asv_bench/"; then
+          cd asv_bench
+          asv machine --yes
+          ASV_OUTPUT="$(asv dev)"
+          if [[ $(echo "$ASV_OUTPUT" | grep "failed") ]]; then
+              echo "##vso[task.logissue type=error]Benchmarks run with errors"
+              echo $ASV_OUTPUT
+              exit 1
+          else
+              echo "Benchmarks run without errors"
+          fi
+      else
+          echo "Benchmarks did not run, no changes detected"
+      fi
+    displayName: 'Running benchmarks'
+    condition: true
```

Comment:
@datapythonista : Why did you add running benchmarks when the original issue was just to migrate `lint.sh` to Azure?  IMO, this should have been considered in a separate PR.

Also, as your changes did not actually modify any Python code, so it was not possible to review how it would look in the Azure logs.  This log here is unfortunately impossible to read:

https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=4580

Reply:
Also, it's somewhat debatable as to whether we even want to just run benchmarks when there's only a diff in `asv_bench`.  We often ask people to run `asv` whenever there are changes to C and Cython code in general, just to make sure that no benchmarks were affected.